### PR TITLE
Use react-volume-viewer library

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
Removes local files and adds the `react-volume-viewer` to the project

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,42 +1,55 @@
...
+};
 
 export default function App() {
   const [controlsVisible, setControlsVisible] = useState(false);
 
+  const [selection, setSelection] = useState({
+    season: BUTTONS.season[0],
+    tide: BUTTONS.tide[0],
+    measurement: BUTTONS.measurement[0],
+  });
```
Comment (mcmcgrath13) [inline src/App.jsx:32]:
```suggestion
  const [selection, setSelection] = useState(Object.keys(BUTTONS).reduce((obj, key) => ({...obj, [key]: BUTTONS[key][0]}), {});
```
A bit less readable, but it's one less place to update if the variables change.   Might be worth making the variables their own `const` then also using that to construct `BUTTONS` (would also obviate the need for the `Object.keys` call here.

File: src/App.jsx [hunk 2]
```diff
@@ -1,42 +1,55 @@
...
-            toggleControls={() => setControlsVisible(!controlsVisible)}
-          />
-          <VolumeViewerWrapper controlsVisible={controlsVisible} />
-        </ControlsProvider>
-      </SelectorProvider>
-
+      <InfoText selection={selection} />
+      <ModelSelector
+        BUTTONS={BUTTONS}
+        ALL_COLOR_MAPS={ALL_COLOR_MAPS}
```
Comment (mcmcgrath13) [inline src/App.jsx:41]: Same question as on the volume viewer of props vs consts for these items.

File: src/App.jsx [hunk 3]
```diff
@@ -1,42 +1,55 @@
...
+        selection={selection}
+        setSelection={setSelection}
+        setColorMap={setColorMap}
+        toggleControls={() => setControlsVisible(!controlsVisible)}
+      />
+      <VolumeViewerWrapper
+        ALL_COLOR_MAPS={ALL_COLOR_MAPS}
```
Comment (mcmcgrath13) [inline src/App.jsx:48]: and here

File: src/assets/config.json [hunk 4]
```diff
@@ -34,33 +34,11 @@
   "slices": 55,
   "x_spacing": 2,
   "y_spacing": 2,
-  "z_spacing": 1,
+  "z_spacing": 2,
```
Comment (mcmcgrath13) [inline src/assets/config.json:37]: why the change from 1 to 2?

File: src/components/ModelSelector.jsx [hunk 5]
```diff
@@ -7,31 +7,37 @@ import {
...
   const {
-    state: { selection },
-    dispatch: selectorDispatch,
-  } = useSelectorContext();
+    BUTTONS,
+    ALL_COLOR_MAPS,
+    selection,
+    setSelection,
+    setColorMap,
+    toggleControls,
```
Comment (mcmcgrath13) [inline src/components/ModelSelector.jsx:17]: move to function signature?

File: src/components/ModelSelector.jsx [hunk 6]
```diff
@@ -7,31 +7,37 @@ import {
...
+  } = props;
 
-  const { dispatch: controlsDispatch } = useControlsContext();
+  function handleMeasurementChange(val) {
+    setSelection({
+      ...selection,
+      measurement: val,
+    });
```
Comment (mcmcgrath13) [inline src/components/ModelSelector.jsx:24]: should this be a function that takes `selection` as an arg?

File: src/components/ModelSelector.jsx [hunk 7]
```diff
@@ -7,31 +7,37 @@ import {
...
-      payload:
-        val.value === "salt"
-          ? colorMaps.find((m) => m.name === "Haline")
-          : colorMaps.find((m) => m.name === "Thermal"),
+  }
+
+  function handleTideChange(val) {
+    setSelection({
+      ...selection,
+      tide: val,
```
Comment (mcmcgrath13) [inline src/components/ModelSelector.jsx:40]: maybe make all of these an object with the tide/season/measurement as keys and the values as these functions, then you can make the variables a map in the jsx as well instead of having repetitive code there

**Review by mcmcgrath13 — COMMENTED** [review]
so many things deleted!! amazing!!

File: src/App.jsx (hunk 1 above)
Comment (RobertGemmaJr) [reply src/App.jsx]: Hmm.. maybe make it better in the config.json?

File: src/App.jsx (hunk 2 above)
Comment (RobertGemmaJr) [reply src/App.jsx:45]: Actually I think I'll make a `constants.js` file like with volume viewer. I should still use the config.json file though, right? That's potentially something that could be automated down the line?

Comment (mcmcgrath13) [reply src/App.jsx:45]: I think the difference in this case is largely perception.  When I see a constants file, I think "I shouldn't touch that", when I see a config file, I think "ooooh - let's go fiddle with that".

Comment (RobertGemmaJr) [reply src/App.jsx]: Yeah I'm inclined to agree. Right now I'm using constants to parse everything in the config files for the app. Constants can pretty much be left untouched as the app changes

File: src/App.jsx (hunk 1 above)
Comment (RobertGemmaJr) [reply src/App.jsx]: I decided to leave it as `Object.keys` but used the same logic in `ModelSelector.jsx` to generate the button groups

File: src/assets/config.json (hunk 4 above)
Comment (RobertGemmaJr) [reply src/assets/config.json]: Whoops just a typo, it's fixed

File: src/components/ModelSelector.jsx (hunk 6 above)
Comment (RobertGemmaJr) [reply src/components/ModelSelector.jsx]: I was missing `selection` but it's a callback not an argument to `setSelection`. `React.useState()` updates state variables asynchronously - adding the callback in `setSelection` forces React to finish any updates before executing the function

```js
setSelection((selection) => ({
      ...selection,
      [button]: value,
}));
```

File: src/components/ModelSelector.jsx (hunk 7 above)
Comment (RobertGemmaJr) [reply src/components/ModelSelector.jsx]: I did the map based off `BUTTONS`

File: src/components/ModelSelector.jsx [hunk 8]
```diff
@@ -41,66 +31,25 @@ export default function ModelSelector({ toggleControls }) {
...
-            {tide.map((m) => {
-              return (
-                <ToggleButton key={m.name} value={m}>
-                  {m.name}
-                </ToggleButton>
-              );
-            })}
-          </ToggleButtonGroup>
-        </Col>
+        {Object.keys(BUTTONS).map((buttonGroup) => (
```
Comment (mcmcgrath13) [inline src/components/ModelSelector.jsx:35]:
```suggestion
        {Object.entries(BUTTONS).map(([group, options]) => (
```

or similar.  This saves the need for a second access to `BUTTONS` on line 43

**Comment by RobertGemmaJr** [comment]
Somewhat unrelated but should we separate this repo from the original fork now that that branch has been archived?

File: src/components/VolumeViewerWrapper.jsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import { Container } from "react-bootstrap";
+import { VolumeViewer } from "react-volume-viewer";
+import {
+  ALL_COLOR_MAPS,
+  MODEL_CONSTANTS,
+  MODEL_DATA,
+} from "../constants/constants";
+
+export default function VolumeViewerWrapper(props) {
+  const { colorMap, selection, controlsVisible } = props;
```
Comment (mcmcgrath13) [inline src/components/VolumeViewerWrapper.jsx:10]:
```suggestion
export default function VolumeViewerWrapper({ colorMap, selection, controlsVisible }) {
```

File: src/components/VolumeViewerWrapper.jsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+  const model = {
+    ...MODEL_CONSTANTS,
+    path: `./assets/models/${selection.season.value}-${selection.tide.value}-${selection.measurement.value}.png`,
+    range:
+      MODEL_DATA[
+        `${selection.season.value}-${selection.tide.value}-${selection.measurement.value}`
```
Comment (mcmcgrath13) [inline src/components/VolumeViewerWrapper.jsx:17]: would it be worth pulling this concatenation out above this `model` declaration as it's used also in path?

**Comment by mcmcgrath13** [comment]
> Somewhat unrelated but should we separate this repo from the original fork now that that branch has been archived?

Probably, but I don't think there's an easy way to do that besides deleting the repo and re-starting with a fresh upload of the git history. Then we'd lose all of the issues and things too...

File: src/components/VolumeViewerWrapper.jsx (hunk 10 above)
Comment (RobertGemmaJr) [reply src/components/VolumeViewerWrapper.jsx]: Yeah that's an easy fix, called it `fileName`

File: src/components/ModelSelector.jsx (hunk 5 above)
Comment (RobertGemmaJr) [reply src/components/ModelSelector.jsx]: Done (:

**Review by mcmcgrath13 — APPROVED** [review]
🙌
